cibuildwheel/util.py

```diff
@@ -81,18 +82,21 @@ def download(url, dest):
         response.close()
 
 
+DependencyConstraints_T = TypeVar('DependencyConstraints_T', bound='DependencyConstraints')
+
+
 class DependencyConstraints:
-    def __init__(self, base_file_path):
+    def __init__(self, base_file_path: str):
         assert os.path.exists(base_file_path)
         self.base_file_path = os.path.abspath(base_file_path)
 
     @classmethod
-    def with_defaults(cls):
+    def with_defaults(cls: Type[DependencyConstraints_T]) -> DependencyConstraints_T:
```

Comment:
We don't have to worry about subclassing, maybe lets just make this a staticmethod? Or do-

```suggestion
    def with_defaults(cls: Type[DependencyConstraints]) -> DependencyConstraints:
```

Reply:
OK, let's go for `@staticmethod`, then?
I know I overdid it a little bit, here. But just wanted do get `--strict` to pass, and worry about what was too much later :-)

Reply:
Yep staticmethod is good

Reply:
Should be done, already (but GitHub shows an old version if you click the `cibuildwheel/util.py` here).